File: String/char_occuring.cpp
// Occuring number of character within a string - O(N) complexity

#include <bits/stdc++.h>
using namespace std;

class Solution{
public:
	static int occur(string &s, char c){
		unordered_map<char,int> hash;
		for(auto &x : s)
			hash[x]++;
        for(auto &y : hash)
			if(y.first==c)
				return y.second;
        return 0;
	}
};

int main(){
    ios_base::sync_with_stdio(0);
    string test = "agoraphobia";
    char c = 'a';
    cout << Solution::occur(test,c);
}

File: String/char_occuring.py
# Occuring number of character within a string - O(N) complexity

from collections import Counter

class Solution(object):
    def occur(self,s:str,c:chr)->int:
        self.freqDict = Counter(s)
        for (self.key,self.val) in self.freqDict.items():
            if(self.key == c):
                return self.val

        return 0 # if there is no character equal to c

if __name__=='__main__':
    test = "agoraphobia"
    s = Solution()
    print(s.occur(test,"a")) # 3
